src/si/bus.stanza

```diff
@@ -0,0 +1,174 @@
+#use-added-syntax(jitx)
+defpackage jsl/si/bus :
+  import core
+  import collections
+  import jitx
+  import jitx/commands
+
+  import jsl/errors
+  import jsl/utils
+  import jsl/design/introspection
+  import jsl/si/helpers
+
+doc: \<DOC>
+A wrapper for `KeyValue` indicating that the given pair of
+objects should be connected by a short trace.
+
+In this toy example, port `p1` will be connected to `r1.p[1]` with
+a short trace.
+```stanza
+require p1 from mcu
+inst r1 : my-resitor
+port p2
+topo-bus(ShortTrace(p1 => r1) => p2)
+```
+<DOC>
+public deftype ShortTrace <: KeyValue
+public defn ShortTrace (kv:KeyValue) :
+  new ShortTrace :
+    defmethod key (this) : key(kv)
+    defmethod value (this) : value(kv) 
+defmethod print (o:OutputStream, st:ShortTrace) :
+  print(o, "ShortTrace(%_ => %_)" % [key(st) value(st)])
+
+doc: \<DOC>
+  This function will construct chained topology net consisting of
+  a sequence of "bus nodes".
+
+  Given a sequence of key-value pairs of nodes, this function
+  will connect them and generate topology segments.
+
+  In this context, a node is either:
+  * A tuple of single-pin or array ports
+  * A single-pin or array port
+  * An instance
+  * An Instantiable
+  * A "ShortTrace"
+
+  If given an Instantiable for one node of the chain, this function will 
+  create as many concrete instances as necessary.
+
+  The left- and right-most nodes of the chain must be pins from the JITX
+  design. Both concrete and abstract require pins work. They cannot be
+  instances or instantiables.
+
+  We assume that all concrete segments are the same size and shape:
+  we can only compute `topo-bus([a, b, c], [d, e, f])` when port `a`
```

Comment:
```
topo-bus([a, b, c], [d, e, f])
```

Did you mean to use a `,` here. Or should it be a `=>`?

Reply:
good catch